nexpose/nexpose.py

```diff
@@ -620,15 +622,20 @@ def RequestReportDelete(self, report_id, reportconfiguration_id=0):
         else:
             return self.ExecuteBasicOnReport("ReportDeleteRequest", report_id)
 
-    def RequestReportAdhocGenerate(self, id):
+    def RequestReportAdhocGenerate(self, id, format='raw-xml-v2', template_id='audit-report'):
         request = """
-<AdhocReportConfig format="raw-xml-v2" template-id="audit-report">
+<AdhocReportConfig format="{format}" template-id="{template_id}">
 <Filters>
-<filter type="scan" id="{0}" />
+<filter type="scan" id="{scan_id}" />
```

Comment:
where does `scan_id` come from?

Reply:
It comes from `id` as part of the method signature. It's always been there, I just renamed it for clarity.